src/move_ref.rs

```diff
@@ -0,0 +1,397 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Move references, for emulating `&move`.
+
+#![allow(missing_docs, unused)]
+
+use core::marker::Unpin;
+use core::mem;
+use core::mem::MaybeUninit;
+use core::ops::Deref;
+use core::ops::DerefMut;
+use core::pin::Pin;
+use core::ptr;
+
+#[cfg(doc)]
+use alloc::{boxed::Box, rc::Rc, sync::Arc};
+
+/// A library emulation of the theoretical `&move T` type.
+///
+/// A `MoveRef<'a, T>` represents a unique reference to `T` for the lifetime
+/// `'a`. Unlike an `&mut T`, though, a `MoveRef<T>` is the *longest-lived* such
+/// reference, entitling it to run `T`'s destructor. In other words,
+/// `MoveRef<T>` owns its referent, but not its referent's storage.
+/// See the [module docs][`crate::move_ref`] for more information.
+///
+/// The main mechanism for obtaining `MoveRef`s is the [`move_ref!()`] macro,
```

Comment:
Documentation is still referring to `move_ref!`, which doesn't exist. Did you mean `moveit!`?